ipfs-api-prelude/src/response/pubsub.rs

```diff
@@ -27,15 +30,60 @@ pub type PubsubPubResponse = ();
 
 #[derive(Debug, Deserialize)]
 pub struct PubsubSubResponse {
-    pub from: Option<String>,
-    pub data: Option<String>,
-    pub seqno: Option<String>,
+    pub from: String,
+
+    #[serde(deserialize_with = "deserialize_data_field")]
+    pub data: Vec<u8>,
+
+    #[serde(deserialize_with = "deserialize_seqno_field")]
+    pub seqno: u64,
+
+    #[serde(rename = "topicIDs", deserialize_with = "deserialize_topic_field")]
+    pub topic_ids: Vec<String>,
+}
+
+fn deserialize_data_field<'de, D>(deserializer: D) -> Result<Vec<u8>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let data: &str = Deserialize::deserialize(deserializer)?;
+
+    let (_, data) = decode(data).expect("Multibase Decoding");
+
+    Ok(data)
+}
+
+fn deserialize_seqno_field<'de, D>(deserializer: D) -> Result<u64, D::Error>
```

Comment:
There is a `serde` file where I have a few other deserializae methods. Could you move these there?